pkg/config/setup/config.go

```diff
@@ -1555,6 +1555,9 @@ func logsagent(config pkgconfigmodel.Setup) {
 	// Add a tag to logs that are truncated by the agent
 	config.BindEnvAndSetDefault("logs_config.tag_truncated_logs", false)
 
+	// Number of logs pipeline instances. Defaults to number of logical CPU cores as defined by GOMAXPROCS.
+	config.BindEnvAndSetDefault("logs_config.pipelines", runtime.GOMAXPROCS(0))
```

Comment:
@gh123man Would this have similar issues as the count of forwarders on high core-count servers (e.g. 96+)? IIRC, we generally have ended up needing to do logarithmic scaling with core counts in other parallelized code (workers, forwarders) to avoid problems.

Reply:
Been chatting with @scottopell about this. 
I am going to run some benchmarks (idle agent collecting no logs with a large number of pipelines) and compare it to the baseline/current implementation to see what the expected overhead is.